### DISCUSSION TITLE
macOS: cat /dev/urandom crashes Ghostty

### DISCUSSION THREAD

**Discussion by bluekeybo** [ghostty-org/ghostty#4008] [Feature Requests, Ideas] [answered]
This is an odd issue, as it doesn't happen all the time. However, once every 5 runs, after quitting and reopening Ghostty, running `cat /dev/urandom` will cause an immediate crash.

Version: 1.0.0
Build: 8469

Thank you!

**Comment by tristan957** [comment]
This isn't useful unless you can provide a stacktrace. It is known that Ghostty is not crash proof when splatting random bytes to the screen.

**Reply by bluekeybo** [reply]
What's the best way to provide the stacktrace? Should I follow what's listed in the Readme to send the crash report?
`SENTRY_DSN=https://e914ee84fd895c4fe324afa3e53dac76@o4507352570920960.ingest.us.sentry.io/4507850923638784 sentry-cli send-envelope --raw <path to ghostty crash>`

**Reply by mitchellh** [reply]
Yes or build a debug build and get one. 

Additionally the bytes that generate the crash would be useful.

I'll close this but reopen with that info!

**Reply by bluekeybo** [reply]
> Yes or build a debug build and get one.
> 
> Additionally the bytes that generate the crash would be useful.
> 
> I'll close this but reopen with that info!

I've narrowed it down further. A crash example is trying to `cat` a file that contains the `ESC` character followed by the letter `c`.

So here's the binary and hexdump of such a file that causes a crash:

`b"\x1bc"`

`0000000    631b`

Crash file:
[crash.txt](https://github.com/user-attachments/files/18277525/crash.txt)

Thank you!

**Reply by mitchellh** [reply]
This does not crash for me.

**Reply by bluekeybo** [reply]
hmmm I did have it once where it didn't crash for me too, but the letter `c` also wasn't printed. I quit Ghostty, opened it again, and the crash came back. Very odd

**Reply by bluekeybo** [reply]
> This does not crash for me.

Just tried with Version 1.0.1 to double-check, and I still get a crash when cat-ing the file above. I've submitted the Sentry crash report. Thank you!

**Reply by bluekeybo** [reply]
Still crashes on `1.1.0` using the file above

**Reply by tristan957** [reply]
Why not provide a backtrace which is more actionable?

**Reply by bluekeybo** [reply]
> I've narrowed it down further. A crash example is trying to `cat` a file that contains the `ESC` character followed by the letter `c`.
> 
> So here's the binary and hexdump of such a file that causes a crash:
> 
> `b"\x1bc"`
> 
> `0000000 631b`
> 
> Crash file: [crash.txt](https://github.com/user-attachments/files/18277525/crash.txt)
> 
> Thank you!

Crash report sent, using Sentry

**Reply by bluekeybo** [reply, marked as answer]
Happy to report that the following version has fixed this issue:

Version: 1.2.0
Build: 12119

Thank you!
